mev-relay-rs/src/relay.rs

```diff
@@ -525,10 +550,12 @@ impl BlindedBlockProvider for Relay {
             let local_header = auction_context.signed_builder_bid.message.header();
             if let Err(err) = validate_header_equality(local_header, execution_payload_header) {
                 warn!(%err, %auction_request, "invalid incoming signed blinded beacon block");
-                return Err(RelayError::InvalidSignedBlindedBeaconBlock.into())
+                return Err(RelayError::InvalidSignedBlindedBeaconBlock.into());
             }
         }
 
+        assert!(self.verify_blinded_block_signature(signed_block).is_ok());
```

Comment:
so if we just `assert` here and the signature is incorrect, the relay would just panic instead of e.g. returning an error to the builder (following the relay APIs spec)...

instead, we should call this function and can prolly just use the `?` operator to bail early if there is a problem